ansible/roles/ci/code_style_check/catkin/files/catkin_lint_xunit.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+
+# Copyright 2014-2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+import time
+from xml.sax.saxutils import escape
+from xml.sax.saxutils import quoteattr
+
+
+def main(argv=sys.argv[1:]):
+    
+    parser = argparse.ArgumentParser(
+        description='Check a package using catkin_lint.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        'paths',
+        nargs='*',
+        default=[os.curdir],
+        help='The package or folder with packages to check with catkin_lint')
+    # not using a file handle directly
+    # in order to prevent leaving an empty file when something fails early
+    parser.add_argument(
+        '--xunit-file',
+        help='Generate a xunit compliant XML file')
+    parser.add_argument(
+        '--lintignore',
+        help='Name of the lintignore file for catkin (normally .catkin_lint_ignore)')
+    args = parser.parse_args(argv)
+
+    if args.xunit_file:
+        start_time = time.time()
+
+    packages = args.paths
+    if not packages:
+        print('No packages found', file=sys.stderr)
+        return 1
+    packages = [os.path.abspath(f) for f in packages]
+
+    catkinlint_bin = shutil.which('catkin_lint')
+    if not catkinlint_bin:
+        return "Could not find 'catkin_lint' executable"
+
+    report = []
```

Comment:
probably is gonna go in a function, so define at top of the function. In theory, it would be good to have variables at top of functions. So it is clear reads which variables are going to get used

Reply:
ok, report=[] defined at the top of a function